server/jetstream_api.go

```diff
@@ -27,118 +28,121 @@ import (
-	JetStreamListStreams = "$JS.STREAM.LIST"
+	// JSApiStreamInfo is for obtaining general information about a named stream.
+	// Will return JSON response.
+	JSApiStreamInfo  = "$JS.API.STREAM.INFO.*"
+	JSApiStreamInfoT = "$JS.API.STREAM.INFO.%s"
 
-	// JetStreamStreamInfo is for obtaining general information about a named stream.
+	// JSApiDeleteStream is the endpoint to delete streams.
 	// Will return JSON response.
-	JetStreamStreamInfo  = "$JS.STREAM.*.INFO"
-	JetStreamStreamInfoT = "$JS.STREAM.%s.INFO"
+	JSApiStreamDelete  = "$JS.API.STREAM.DELETE.*"
+	JSApiStreamDeleteT = "$JS.API.STREAM.DELETE.%s"
 
-	// JetStreamDeleteStream is the endpoint to delete streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamDeleteStream  = "$JS.STREAM.*.DELETE"
-	JetStreamDeleteStreamT = "$JS.STREAM.%s.DELETE"
+	// JSApiPurgeStream is the endpoint to purge streams.
+	// Will return JSON response.
+	JSApiStreamPurge  = "$JS.API.STREAM.PURGE.*"
+	JSApiStreamPurgeT = "$JS.API.STREAM.PURGE.%s"
 
-	// JetStreamPurgeStream is the endpoint to purge streams.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamPurgeStream  = "$JS.STREAM.*.PURGE"
-	JetStreamPurgeStreamT = "$JS.STREAM.%s.PURGE"
+	// JSApiDeleteMsg is the endpoint to delete messages from a stream.
+	// Will return JSON response.
+	JSApiMsgDelete  = "$JS.API.MSG.DELETE.*"
+	JSApiMsgDeleteT = "$JS.API.MSG.DELETE.%s"
 
-	// JetStreamDeleteMsg is the endpoint to delete messages from a stream.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamDeleteMsg  = "$JS.STREAM.*.MSG.DELETE"
-	JetStreamDeleteMsgT = "$JS.STREAM.%s.MSG.DELETE"
+	// JSApiMsgGet is the template for direct requests for a message by its stream sequence number.
+	// Will return JSON response.
+	JSApiMsgGet  = "$JS.API.MSG.GET.*"
+	JSApiMsgGetT = "$JS.API.MSG.GET.%s"
 
-	// JetStreamCreateConsumer is the endpoint to create durable consumers for streams.
-	// You need to include the stream and consumer name in the subject.
-	// Will return +OK on success and -ERR on failure.
-	JetStreamCreateConsumer  = "$JS.STREAM.*.CONSUMER.*.CREATE"
-	JetStreamCreateConsumerT = "$JS.STREAM.%s.CONSUMER.%s.CREATE"
+	// JSApiConsumerCreate is the endpoint to create ephemeral consumers for streams.
+	// Will return JSON response.
+	JSApiConsumerCreate  = "$JS.API.CONSUMER.CREATE.*"
+	JSApiConsumerCreateT = "$JS.API.CONSUMER.CREATE.%s"
 
-	// JetStreamCreateEphemeralConsumer is the endpoint to create ephemeral consumers for streams.
-	// Will return +OK <consumer name> on success and -ERR on failure.
-	JetStreamCreateEphemeralConsumer  = "$JS.STREAM.*.EPHEMERAL.CONSUMER.CREATE"
-	JetStreamCreateEphemeralConsumerT = "$JS.STREAM.%s.EPHEMERAL.CONSUMER.CREATE"
+	// JSApiDurableConsumerCreate is the endpoint to create ephemeral consumers for streams.
+	// You need to include the stream and consumer name in the subject.
+	JSApiDurableCreate  = "$JS.API.DURABLE.CREATE.*.*"
```

Comment:
if I am reading through the list of subjects thinking "lets write ACLs for consumer creation" I will 100% miss this one.

Reply:
I wanted durables to stand out and be able to be treated differently, but true if you wanted to cut off all consumer actions you would need 2 deny clauses. 

I can add it back in and fix up the token parsing code for this handler.